File: 프로그래머스/unrated/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.py
# 오늘날짜
# 유효기간 담은 terms
def days(lst):
    y,m,d = lst
    result = y * 336 + m * 28 + d
    return result

def solution(today, terms, privacies):
    term_dict = {}
    answer = []
    num = 0
    now = days(list(map(int,today.split('.'))))
    
    for term in terms:
        AL, NUM = term.split()
        term_dict.setdefault(AL,int(NUM))
    
    for p in privacies:
        num += 1
        date, al = p.split()
        pri = days(list(map(int,date.split('.'))))
        pri += term_dict.get(al) * 28 - 1
        if pri < now:
            answer.append(num)
    
    return answer

File: 프로그래머스/Lv.1/150370. 개인정보 수집 유효기간/개인정보 수집 유효기간.js
function solution(today, terms, privacies) {
    const [ny, nm, nd] = today.split(".").map(Number)
    const Type = {}
    let result = []
    
    terms.forEach((info,idx) => {
        const [alpha, month] = info.split(" ")
        Type[alpha] = Number(month)
    })
    
    const totalDate = nd + (nm * 28) + (ny * 12 * 28)
    privacies.forEach((info,idx) => {
        const [date, alpha] = info.split(" ")
        const [y, m, d] = date.split(".").map(Number)
        const signDate = d + (m * 28) + (y * 12 * 28)
        
        if (totalDate - signDate >= Type[alpha] * 28) {
            result.push(idx+1)
        }
    })
    
    
    return result;
}